internal/controllers/shared_vehicles.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/DIMO-Network/vehicle-events-api/internal/config"
+	"io"
+	"net/http"
+
+	"github.com/rs/zerolog"
+)
+
+type Vehicle struct {
+	TokenID string `json:"tokenId"`
+}
+
+func GetSharedVehicles(devLicense []byte, logger zerolog.Logger) ([]Vehicle, error) {
+	ethAddress := fmt.Sprintf("0x%x", devLicense)
+
+	identityAPIURL := config.GetSettings().IdentityAPIURL
+
+	query := `
+		query($eth: String!) {
+			vehicles(first: 50, filterBy: { privileged: $eth }) {
+				nodes {
+					tokenId
+				}
+			}
+		}`
+	payload := map[string]interface{}{
```

Comment:
it seems like there is a fair amount of code here that could be refactored into a more general use identity-api wrapper, something like this: https://github.com/DIMO-Network/vehicle-signal-decoding/blob/main/internal/gateways/identity_api.go